@xen-orchestra/backups/runBackupWorker.js

```diff
@@ -0,0 +1,39 @@
+const path = require('path')
+const pDefer = require('promise-toolbox/defer')
+const { createLogger } = require('@xen-orchestra/log')
+const { fork } = require('child_process')
+
+const { warn } = createLogger('xo:backups:backupWorker')
+
+exports.runBackupWorker = function runBackupWorker(params, onLog) {
+  const { promise, resolve, reject } = pDefer()
```

Comment:
You don't need to use `promise-toolbox/defer` here, you can simply use normal promise construction:

```js
return new Promise((resolve, reject) => {
  // …
})
```

This:
- is more idiomatic
- does not depend on external library
- ensure all sync errors are properly caught and wrapped into the promise